docs/source/api/preferences.rst

```diff
@@ -0,0 +1,702 @@
+.. 2022-03-17 by Xylopyrographer. Initial PR release.
+
+###########
+Preferences
+###########
+
+About
+-----
+
+The Preferences library is unique to arduino-esp32. It should be considered as the replacement for the Arduino EEPROM library.
+
+It uses a portion of the on-board non-volatile memory (NVS) of the ESP32 to store data. This data is retained across restarts and loss of power events to the system. 
+
+Preferences works best for storing many small values, rather than a few large values. If large amounts of data are to be stored, consider using a file system library such as LitteFS.
+
+The Preferences library is usable by all ESP32 variants.
+
+
+Header file
```

Comment:
For titles and heading, we use capital letters for the first letter of each word. Exceptions are: is, of, or, for, and...